Runtime/Highlight/DefaultHighlighter.cs

```diff
@@ -0,0 +1,445 @@
+                    transform = meshFilter.transform.localToWorldMatrix
+                };
+
+                combinedInstances.Add(combineInstance);
+            }
+
+            return combinedInstances;
+        }
+        
+        private IEnumerable<CombineInstance> CollectMeshInformationFromSkinnedMeshRenderer(SkinnedMeshRenderer skinnedMeshRenderer)
+        {
+            List<CombineInstance> combinedInstances = new List<CombineInstance>();
+
+            for (int i = 0; i < skinnedMeshRenderer.sharedMesh.subMeshCount; i++)
+            {
+                CombineInstance combineInstance = new CombineInstance
+                {
+                    subMeshIndex = i,
+                    mesh = skinnedMeshRenderer.sharedMesh,
+                    transform = skinnedMeshRenderer.transform.localToWorldMatrix
+                };
+
+                combinedInstances.Add(combineInstance);
+            }
+
+            return combinedInstances;
+        }
+
+        protected void DisableRenders()
+        {
+            if (highlightMeshRenderer != null)
+            {
+                highlightMeshRenderer.enabled = true;
+                highlightMeshRenderer.gameObject.SetActive(true);
+            }
+
+            foreach (Renderer renderer in renderers)
+            {
+                if (renderer != null)
+                {
+                    renderer.enabled = false;
+                }
+            }
+        }
+
+        protected void ReenableRenderers()
+        {
+            if (highlightMeshRenderer != null)
+            {
+                highlightMeshRenderer.enabled = false;
+                highlightMeshRenderer.gameObject.SetActive(false);
+            }
+
+            foreach (Renderer renderer in renderers)
+            {
+                if (renderer != null)
+                {
+                    renderer.enabled = true;
+                }
+            }
+        }
+
+        protected Material NewHighlightMaterial(Color highlightColor)
+        {
+            Material material = CreateHighlightMaterial();
+            material.color = highlightColor;
+
+            // In case the color has some level of transparency,
+            // we set the Material's Rendering Mode to Transparent. 
+            if (Mathf.Approximately(highlightColor.a, 1f) == false)
+            {
+                material.SetInt("_SrcBlend", (int)BlendMode.One);
+                material.SetInt("_DstBlend", (int)BlendMode.OneMinusSrcAlpha);
+                material.SetInt("_ZWrite", 0);
+                material.DisableKeyword("_ALPHATEST_ON");
+                material.DisableKeyword("_ALPHABLEND_ON");
+                material.EnableKeyword("_ALPHAPREMULTIPLY_ON");
+                material.renderQueue = 3000;
+            }
+            
+            return material;
+        }
+        
+        protected Material NewHighlightMaterial(Texture mainTexture)
+        {
+            Material material = CreateHighlightMaterial();
```

Comment:
should create the material with the shader from GetShader

Reply:
Same as above